src/app/src/components/ClaimedFacilitiesDetails.jsx

```diff
@@ -1,9 +1,543 @@
+                    variant="title"
+                    style={claimedFacilitiesDetailsStyles.headingStyles}
+                >
+                    Point of contact
+                    <span
+                        style={
+                            claimedFacilitiesDetailsStyles.switchSectionStyles
+                        }
+                    >
+                        <Switch
+                            color="primary"
+                            onChange={updateContactVisibility}
+                            checked={data.point_of_contact_publicly_visible}
+                        />
+                        Publicly visible
+                    </span>
+                </Typography>
+                <InputSection
+                    label="Contact person name"
+                    value={data.point_of_contact_person_name}
+                    onChange={updateContactPerson}
+                    disabled={updating}
+                />
+                <InputSection
+                    label="Email"
+                    value={data.point_of_contact_email}
+                    onChange={updateContactEmail}
+                    disabled={updating}
+                />
+                <Typography
+                    variant="headline"
+                    style={claimedFacilitiesDetailsStyles.headingStyles}
+                >
+                    Office information
+                    <span
+                        style={
+                            claimedFacilitiesDetailsStyles.switchSectionStyles
+                        }
+                    >
+                        <Switch
+                            color="primary"
+                            onChange={updateOfficeVisibility}
+                            checked={data.office_info_publicly_visible}
+                        />
+                        Publicly visible
+                    </span>
+                </Typography>
+                <InputSection
+                    label="Office name"
+                    value={data.office_official_name}
+                    onChange={updateOfficeName}
+                    disabled={updating}
+                />
+                <InputSection
+                    label="Address"
+                    value={data.office_address}
+                    onChange={updateOfficeAddress}
+                    disabled={updating}
+                />
+                <InputSection
+                    label="Country"
+                    value={data.office_country_code}
+                    onChange={updateOfficeCountry}
+                    disabled={updating}
+                    isSelect
+                    selectOptions={countryOptions}
+                />
+                <InputSection
+                    label="Phone number"
+                    value={data.office_phone_number}
+                    onChange={updateOfficePhone}
+                    disabled={updating}
+                />
+                {errorUpdating && (
```

Comment:
Consider opening an issue to use `full_clean` on the model to trigger Django's model validation and generate helpful error messages that we can show to the claimant. We do this when parsing uploaded files https://github.com/open-apparel-registry/open-apparel-registry/blob/8fdc09c4d6cac5f4aefc3a313e9c04d84e5ab49f/src/django/api/processing.py#L127

We are currently returning 500 when the claimant enters too many characters into a field.

<img width="1680" alt="Screen Shot 2019-06-12 at 8 18 06 PM" src="https://user-images.githubusercontent.com/17363/59401488-b62f5980-8d4f-11e9-8a66-f40be34deae3.png">

We could also make use of `maxlength` to prevent these types of errors.

Reply:
I made https://github.com/open-apparel-registry/open-apparel-registry/issues/589 for handling it. I think we could also use the existing serializer to check somehow.